src/mscorlib/shared/System/Memory.cs

```diff
@@ -163,6 +160,30 @@ private Memory(object obj, int index, int length)
         /// </summary>
         public bool IsEmpty => _length == 0;
 
+        /// <summary>
+        /// For <see cref="Memory{Char}"/>, returns a new instance of string that represents the characters pointed to by the memory.
+        /// Otherwise, returns a <see cref="string"/> with the name of the type and the number of elements.
+        /// </summary>
+        public override string ToString()
+        {
+            if (typeof(T) == typeof(char))
+            {
+                if (_object is string str)
```

Comment:
Why can't this be just:

```
if (typeof(T) == typeof(char))
{
    return Span.ToString();
}
```

?

Reply:
I thought of doing that (I do that for the OwnedMemory case anyway), but in the case of string, we would create a span from the string, slice it, and and then grab the string again. Seemed unnecessary.

Similarly, with array, we would construct the span around the array, pin it, and create the new string. It was simpler to just create the substring directly from the array.

Span.ToString() would be a clean implementation though, if you think that is the way to go.

Reply:
> in the case of string, we would create a span from the string, slice it, and and then grab the string again.

Ok, this make sense as an optimization.

> t was simpler to just create the substring directly from the array.

This is not saving anything meaningful. Constructing Span from array is cheap. So what about something like:

```
if (typeof(T) == typeof(char))
{
     return (_object is string str) ? str.Substring(_index, _length)  : Span.ToString();
}
```

Reply:
> This is not saving anything meaningful. Constructing Span from array is cheap.

Yes, that works. I will add and run tests on the corefx side to ensure correctness.

Reply:
What about this extra check that we end up doing when _object is a char[] if we call Span.ToString() instead?
https://github.com/dotnet/coreclr/blob/master/src/mscorlib/shared/System/Memory.cs#L215

Reply:
It costs nothing. You can always measure it if you are not sure.

Reply:
Yep, just measured. It has no impact.